File: tree_py/binarytreedisp/valueutil.py
class ValueUtil:
    #
    #
    def __init__(self):
        self.__float_precision = 2

    #
    #
    def get_float_precision(self):
        return self.__float_precision

    #
    #
    def set_float_precision(self, precision: int):
        if type(precision) is not int or precision < 0:
            raise ValueError('Invalid argument: precision must be non-negative integer')

        self.__float_precision = precision

    #
    #
    def get_str(self, value):
        type_value = type(value)

        if type_value is float:
            return str(round(value, self.__float_precision))

        if type_value is str:
            return value

        return str(value)

File: bintree_java/src/binarytreedisp/ValueUtil.java
package binarytreedisp;


import java.text.NumberFormat;



public class ValueUtil {
    private int floatPrecision = 2;
    private NumberFormat nf = NumberFormat.getInstance();



    public ValueUtil() {
        this(2);
    }



    public ValueUtil(int floatPrecision) {
        setFloatPrecision(floatPrecision);
    }



    public int getFloatPrecision() {
        return floatPrecision;
    }



    public void setFloatPrecision(int precision) {
        if (precision < 0)
            throw new IllegalArgumentException("float precision must be a non-negative integer");

        this.floatPrecision = precision;
        nf.setMaximumFractionDigits(precision);
    }



    public String getStr(int value) {
        return String.valueOf(value);
    }

    public String getStr(long value) {
        return String.valueOf(value);
    }

    public String getStr(float value) {
        //return String.valueOf(value);
        return nf.format(value);
    }

    public String getStr(double value) {
        //return String.valueOf(value);
        return nf.format(value);
    }

    public String getStr(char value) {
        return String.valueOf(value);
    }

    public String getStr(boolean value) {
        return String.valueOf(value);
    }

    public String getStr(Object value) {
        return String.valueOf(value);
    }
}
